src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -440,11 +440,16 @@ private void AddLexicalClassifications(IClassificationService classificationServ
 
             using var _ = Classifier.GetPooledList(out var tempList);
 
-            // If we have a syntax root ready, use the direct, non-async/non-blocking approach to getting classifications.
-            if (!lastProcessedDocumentOrRoot.TryGetFirst(out var root))
-                classificationService.AddSyntacticClassificationsAsync(lastProcessedDocumentOrRoot.Second, span.Span.ToTextSpan(), tempList, cancellationToken).Wait(cancellationToken);
```

Comment:
calling AddSyntacticClassificationsAsync meant the callee did a GetSyntaxRootAsync, which we then just .Wait'ed on here.

We have an alternative to that for our own use for exactly these scenarios: `GetSyntaxRootSynchronously`.  This call does the work to compute the root on the current thread, not going through any async paths. Switched to using the below.

In all cases, this makes the current UI thread wait until the root is ready.  However, with the new code we're not dependent on how well the threadpool is doing, and any extra delays caused by going through that.